graphql/util.go

```diff
@@ -1290,3 +1290,67 @@ func getAPISubscriptionsForProject(ctx context.Context, projectId string) ([]*re
 	}
 	return res, nil
 }
+
+// GroupProjects takes a list of projects and groupes them by their repo. If onlyDefaultedToRepo is true,
+// it only groups projects that are defaulted to the repo
+func GroupProjects(projects []model.ProjectRef, onlyDefaultedToRepo bool) ([]*GroupedProjects, error) {
+	groupsMap := make(map[string][]*restModel.APIProjectRef)
+
+	for _, p := range projects {
+		groupName := fmt.Sprintf("%s/%s", p.Owner, p.Repo)
+
+		if onlyDefaultedToRepo && p.UseRepoSettings {
+			groupName = ""
+		}
+
+		apiProjectRef := restModel.APIProjectRef{}
+		if err := apiProjectRef.BuildFromService(p); err != nil {
+			return nil, errors.Wrap(err, "error building APIProjectRef from service")
+		}
+
+		if projs, ok := groupsMap[groupName]; ok {
+			groupsMap[groupName] = append(projs, &apiProjectRef)
+		} else {
+			groupsMap[groupName] = []*restModel.APIProjectRef{&apiProjectRef}
+		}
+	}
+
+	groupsArr := []*GroupedProjects{}
+
+	for groupName, groupedProjects := range groupsMap {
+		gp := GroupedProjects{
+			Name:             &groupName, //deprecated
+			GroupDisplayName: groupName,
+			Projects:         groupedProjects,
+		}
+
+		if groupedProjects[0].UseRepoSettings {
+			repoRef, err := model.FindOneRepoRef(utility.FromStringPtr(groupedProjects[0].RepoRefId))
```

Comment:
Should check nil

Reply:
Sorry I meant we should check if repoRef is nil (maybe just logging a warning idk)

Reply:
done